docs/frontend_architecture/HTML5_API.rst

```diff
@@ -0,0 +1,93 @@
+Custom Navigation
+=================
+
+The purpose of the ``kolibri.js`` extension of our HTML5 API is to allow a sandboxed HTML5 app to safely request the main Kolibri application's data.
+
+External/partner product teams can create HTML5 applications that are fully embeddable within Kolibri and can read Kolibri content data, which they otherwise wouldn't be able to access. This opens up possibilities for creative ways in which learners can engage with content, because partners can create any type of app they want. The app could be something completely new, developed for a content source that we are adding to the platform, or it could be a branded, offline recreation of a partner's existing learning app that previously would not have been able to exist on Kolibri.
+
+When a user has permissions to access a custom channel, and they click on it in the main learn tab, rather than viewing "normal Kolibri," they will experience a full-screen HTML5 app. One "out-of-the-box" user interaction is the "navigateTo()" function, which opens  a modal that displays a content node. For other data fetching requests, the app, not Kolibri, has the responsibilty of determining what to do with that data.
+
+
+Basic API
+~~~~~~~~~
+
+
+Access the API from within an HTML5 app by using ``window.kolibri.[function]``
+
+Functions:
+
+.. code-block:: javascript
+
+    /*
+    * Method to query contentnodes from Kolibri and return
+    * an array of matching metadata
+    * @param {Object} options - The different options to filter by
+    * @param {string=} options.parent - id of the parent node to filter by, or 'self'
```

Comment:
I just noticed that been using a different "optional parameter" syntax in my JSDoc strings for the theming. Personally, I like the style I use more, since it's emphasizing the argument name, rather than the type, e.g.
```suggestion
    * @param {string} [options.parent] - id of the parent node to filter by, or 'self'
```